common/txmgr/address_state.go

```diff
@@ -0,0 +1,257 @@
+package txmgr
+
+import (
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"gopkg.in/guregu/null.v4"
+)
+
+// AddressState is the state of all transactions for a given address
+type AddressState[
+	CHAIN_ID types.ID,
+	ADDR, TX_HASH, BLOCK_HASH types.Hashable,
+	R txmgrtypes.ChainReceipt[TX_HASH, BLOCK_HASH],
+	SEQ types.Sequence,
+	FEE feetypes.Fee,
+] struct {
+	lggr        logger.SugaredLogger
+	chainID     CHAIN_ID
+	fromAddress ADDR
+
+	sync.RWMutex
+	idempotencyKeyToTx map[string]*txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]
+	unstarted          *TxPriorityQueue[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]
```

Comment:
nit on the naming here. I suggest we specify in the name whether these are sets of transactions (Queues, maps, etc) or an individual transaction (such as inprogress). For example:

unstarted -> unstartedTxns
inprogress -> inprogressTxn or currentTxn or something

It's not a big deal but it will make the code more readable.